### ISSUE TITLE
Does the Definition of "Dragging Movements" Intentionally Limit the Scope to Moving Objects/Items?

### ISSUE THREAD

**Issue by SuzanneTaylor** [w3c/wcag#1560]
The definition of "Dragging Movements" appears to exclude when a user defines a path of their own (not content-defined) by dragging, rather than moving an object by dragging. Is this intentional or an oversight? 

Path-based gestures, where the _content_ defines a path, are covered under WCAG 2.1 SC 1.5.1. 

And, until you read the definition of Dragging Movements, WCAG 2.2 SC 2.5.7 sounds like it covers any dragging movement, including:

- leaving breadcrumbs
- defining a path that items will follow later (as in a strategy game)
- white board drawings, where precise pixel-by-pixel line drawing is unnecessary
- press and hold and move through space to collect items in that space (as in harvesting a field)
- press and hold to draw a selection rectangle
- press and hold to apply color to a shape
- press and hold to connect representations of wires, where additional wire appears as you drag, rather than an animation that seems like the user is moving the wire

The definition does seem to cover cases where you press and hold to, for example, lead a character through a maze.

**Comment by detlevhfischer** [comment]
I think the first four bullet points (though not sure what 'leaving breadcrumbs' would look like) fall under 2.5.1 Pointer Gestures as it is the actual path that matters, not just start and end point of th edragging movement. For the next two, the exact path probably does not matter, just the start and end point. It looks however that it would not be a simple (single point) "press and hold" which would also fall under 2.5.1, but a "press (get pop-up) and move to endpoint" (e.g. a particular colour tile, or the corner endpoint of a selection rectangle). As to the last in your list (connecting wires) it looks like the path itself would again be relevant as following it would lead to particular wires popping up in the vicinity of the path (if I understand correctly), so this would likely fall under 2.5.1. 
For drawing selection rectangles an picking a colour, we could amend the 2.5.7 Understanding text and add these as examples of dragging movements.

**Comment by SuzanneTaylor** [comment]
I think that under 2.5.1, the first three are "free form paths" and meet the exception, "This success criterion is scoped to path-based gestures as it may be difficult or impossible to provide an alternative for free-form paths." It is possible that 2.5.7 is also not meant to cover these cases, and any handling of these cases would wait for WCAG 3. (In my opinion, that is fine.)

Harvesting a field, yes, could meet 2.5.1 since the there are intermediary points (each item to be harvested) even though the user could chose any order. It is possible that the ability to chose the intermediary points in any order should be mentioned in the understanding document. (This should also not meet 2.5.1 based on a tedious series of clicks, but instead, for example, by being able to click to define rows or rectangles that contain what you want to harvest. )

I'm still concerned that 2.5.7 provides different requirements based on the animation employed. 

### Example 1 Wires: 
For example, imagine you need to press and hold to engage a wire connection point on the left, drag along any path at all that you like, then release the mouse at a wire connection point on the right, to cause a new bit of wire to appear. The definition of Dragging Movement, where something must follow the mouse would not apply: **"an operation where the pointer engages with an element on the down event and the element (or a representation of its position) follows the pointer"**. However, if the same application happened to employ an animation so that it appeared you were dragging a wire-generating tool across, then something would be following the pointer and so the guideline would apply.

### Example 2 Selection Rectangle:
**"an operation where the pointer engages with an element on the down event and the element (or a representation of its position) follows the pointer"**
Imagine that you are dragging a selection rectangle with just your mouse pointer, no object would follow the mouse, and the definition would not be met. But, imagine the very same functionality where you click on a net and drag the net to create a selection rectangle to select everything within, this would apply, since something would be following the mouse pointer.

I agree that listing a couple of examples like this in the understanding document for 2.5.7 would solve the issue, if it is too late to alter the definition of dragging movement. 

If the definition can still be edited perhaps something like:

"an operation where the pointer engages with an element on the down event and the element (or a representation of its position) follows the pointer"
OR
"an operation where the pointer begins creating an item based on the location of the down event and completes generating that item base on the location of the up event"

(uses "based on" rather than "at" to cover situations where a user might be dragging a rectangle, but generating a circle within the rectangle, or something like selecting text, where the selection ends at the end of the closest text)

**Comment by detlevhfischer** [comment]
Hi @SuzanneTaylor 
You suggest adding an OR clause to the definition. I think what you propose to add strays too far from the pointer interaction we are targeting here, which is dragging movements.

If there is no visible representation during dragging (neither a picked-up element nor some other representation, like the end point of a selection rectangle) I doubt that we should talk of dragging, or that this would even fall under dragging.  I also do not think we need to single out creating / generating elements as the result of the dragging motion. The merit of the definition as it stands is that it does not specify what happens as a result of the action. Whether that is sorting, reallocating elements to another group, or the creation of a new element (like in your wire connection example) should not matter. if we follow your suggestion, I fear other potential use cases will crop up that would need to be added. IMO, we cannot enumerate / cover all possible variants of the *results* of dragging but should stick to a simple operational description of the pointer interaction itself.

**Comment by SuzanneTaylor** [comment]
In that case, as you mentioned, I think it will be important to have some examples in the understanding document of cases where the animation that accompanies the dragging motion invokes generating something more so than moving something, so that changing the animation (which provides no known benefit) does not become a way to meet WCAG 2.2.

So if a Web application where the user needs to define a rectangle gives you a small rectangle with corners to drag (which clearly meets the definition) or if a Web application instead changes the mouse pointer to cross hairs to allow the very same thing, these two cases should have the same requirements.

**Comment by alastc** [comment]
@detlevhfischer would you be able to outline a couple of examples for the understanding document?

**Comment by detlevhfischer** [comment]
@alastc In principle yes but I think I have not fully understood what the examples should be demonstrating. I do not understand the part

> that accompanies the dragging motion invokes generating something more so than moving something, so that changing the animation (which provides no known benefit) does not become a way to meet WCAG 2.2.

I don't know what "changing the animation" means here, and how that would be seen as "meeting WCAG 2.2" (presumably SC 2.5.7). If the aim is to limit dragging to the object and excluding any events that are caused by the proximity of the things dragged to other things, that may better be formulated in a generic manner rather than by giving examples for what dragging does *not* relate to?

@SuzanneTaylor maybe you can clarify what kind of example you have in mind? Do you want us to spell out that the events triggered in the wiring case do not fall under SC 2.5.7? I would assume if there is another a11y supported way of achieving the same wiring outcome (i.e. one that does not rely on dragging) there would be no reason to implement single point-operable pointer alternatives for this interaction. In the absence of such an alternative, I guess there would need to be a way of simulating the dragging action my single taps/clicks (e.g., via protrusions on the dragged element, or all taps/clicks to the top/bottom/left/right of the element move it stepwise top/bottom/left/right, or whatever).

As to the defined rectangle, I can imagine that a two-step implementation would be possible where you first select a corner and then move that corner stepwise by pointer-activating arrows pointing up/down left/right) (for keyboard operation the same thing could conceivably be done with arrow keys).

**Comment by alastc** [comment]
On reflection, if the interaction does not move an element or create a path, I'm not sure it would be an issue.

For example, if the interaction is to click on one thing, release, move to another point, and click to put it down, that's fine (for this SC).

The focus of this SCs is where you are dragging a thing or making a path with your dragging.

There could be poor implementation of the point-to-point interactions, but that wasn't really the target of this SC.

**Comment by SuzanneTaylor** [comment]
I think a clear/common example that would fall under "Dragging Movements" but where the imagery/animation shown is not exactly moving something would be a dragging motion used to draw a line between items in two different columns in a matching exercise. Imagine the mouse cursor looks like a pencil and you are asked to draw a lines between words and their definitions. No intermediary points are relevant (so it is not "Pointer Gestures) but you do not have the sense that you are moving something. 

(There is a third category, which I think I mixed into the examples above, making this comment confusing. This is the case where there are no required intermediary points, but the intermediary points are still tracked and still have an effect. This happens a lot in games. The user could ideally chose the exact path they want, but even if they can't, they can still enjoy the game.)

**Comment by detlevhfischer** [comment]
@SuzanneTaylor The examples you gave do not seem to be proper examples of a dragging movement since the mouse does not have to stay down (or the fingerpointer in contact wit hthe screen surface) as in the default dragging operation - an initial state change via a simple tap/click that turns the cursor into the corner o a rectangle or endpoint of a connecting line, followed by a second simple tap/click that sets the endpoint (and thereby draws the line or draws the rectangle).

Where intermediary points do have an impact on the outcome, we would not have a dragging movement in the sense that intermediate points do nit matter.

Would the way forward be spelling out cases of rectangles / line connections / game paths in the understanding doc to separate them from dragging movements?

**Comment by alastc** [comment]
The update was agreed today: https://www.w3.org/2021/12/03-ag-minutes.html#t06

Noting that the understanding doc will be further reviewed.

**Comment by SuzanneTaylor** [comment]
> The update was agreed today: https://www.w3.org/2021/12/03-ag-minutes.html#t06
> 
> Noting that the understanding doc will be further reviewed.

Yes, this full address the concern. thank you for considering this
